curriculum/challenges/english/14-responsive-web-design-22/learn-responsive-web-design-by-building-a-piano/612e83ec2eca1e370f830511.md

```diff
@@ -7,48 +7,53 @@ dashedName: step-8
 
 # --description--
 
-Add a `link` element within your `head` element. For that `link` element, set the `rel` attribute to `stylesheet` and the `href` to `./styles.css`.
+Add a `link` element inside your `head` element. Give it a `rel` attribute set to `stylesheet` and an `href` attribute set to `styles.css`.
 
 # --hints--
 
 Your code should have a `link` element.
 
 ```js
-assert.match(code, /<link/)
+assert.isNotNull(document.querySelector('link'));
 ```
 
 You should not change your existing `head` tags. Make sure you did not delete the closing tag.
 
 ```js
-const heads = document.querySelectorAll('head');
-assert.equal(heads?.length, 1);
+const headElements = document.querySelectorAll('head');
+assert.strictEqual(headElements?.length, 1);
 ```
 
-You should have one self-closing `link` element.
+You should have one `link` element.
 
 ```js
-assert(document.querySelectorAll('link').length === 1);
+assert.strictEqual(document.querySelectorAll('link')?.length, 1);
 ```
 
-Your `link` element should be within your `head` element.
+Your `link` element should be inside your `head` element.
 
 ```js
-assert.exists(document.querySelector('head > link'));
+const headContentRegex = /(?<=<head\s*>)(?:.|\s*)*?(?=<\/head\s*>)/;
+const headElementContent = code.match(headContentRegex);
+
+const headElement = document.createElement("head");
+headElement.innerHTML = headElementContent;
+assert.isNotNull(headElement.querySelector('link'));
```

Comment:
This does not make sense to me. Why are we creating our own `head` element and putting the user's content inside?

Reply:
We can't directly test the actual head element using DOM methods.

This passes the test we currently have.

```
<link rel="stylesheet" href="styles.css">
<head>
  <meta charset="UTF-8" />
  <title>Piano</title>
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
</head>
```

We have the same problem in multiple places and need to test for different elements inside the head element. So instead of writing different regexs that matches on different requirements, we write one that just matches on the head content and construct our own head element we can test using DOM methods.